derive/src/lib.rs

```
@@ -0,0 +1,513 @@
+	let options = get_options(&ast.attrs, "ethabi_contract_options")?;
+	let path = get_option(&options, "path")?;
+	let normalized_path = normalize_path(path)?;
+	let source_file = fs::File::open(&normalized_path)
+		.chain_err(|| format!("Cannot load contract abi from `{}`", normalized_path.display()))?;
+	let contract = Contract::load(source_file)?;
+
+	let functions: Vec<_> = contract.functions().map(impl_contract_function).collect();
+	let events_impl: Vec<_> = contract.events().map(impl_contract_event).collect();
+	let constructor_impl = contract.constructor.as_ref().map(impl_contract_constructor);
+	let logs_structs: Vec<_> = contract.events().map(declare_logs).collect();
+	let events_structs: Vec<_> = contract.events().map(declare_events).collect();
+	let func_structs: Vec<_> = contract.functions().map(declare_functions).collect();
+
+	let name = get_option(&options, "name")?;
+	let name = syn::Ident::new(name);
+	let functions_name = syn::Ident::new(format!("{}Functions", name));
+	let events_name = syn::Ident::new(format!("{}Events", name));
+
+	let file_path = string_ident(&normalized_path.display().to_string());
+
+	let result = quote! {
+		use ethabi;
+
+		pub mod logs {
+			use ethabi;
+			#(#logs_structs)*
+		}
+
+		pub mod events {
+			use ethabi;
+
+			#(#events_structs)*
+		}
+
+		pub mod functions {
+			use ethabi;
+
+			#(#func_structs)*
+		}
+
+		pub struct #functions_name<'a> {
+			contract: &'a ethabi::Contract,
+		}
+
+		impl<'a> #functions_name<'a> {
+			#(#functions)*
+		}
+
+		pub struct #events_name<'a> {
+			contract: &'a ethabi::Contract,
+		}
+
+		impl<'a> #events_name<'a> {
+			#(#events_impl)*
+		}
+
+		/// Contract
+		pub struct #name {
+			contract: ethabi::Contract,
+		}
+
+		impl Default for #name {
+			fn default() -> Self {
+				#name::new()
+			}
+		}
+
+		impl #name {
+			pub fn new() -> Self {
+				let contract = ethabi::Contract::load(include_bytes!(#file_path) as &[u8]).unwrap();
```

That's a runtime failure, the JSON file should be compiled into the binary or the method should return a possible error.